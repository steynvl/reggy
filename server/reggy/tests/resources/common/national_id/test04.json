{
  "type": "National ID",
  "information": {
    "kind": "Finnish Social Security Number"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}