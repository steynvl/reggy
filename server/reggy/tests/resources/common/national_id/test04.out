Java
Pattern regex = Pattern.compile("(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]");
(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]
Perl
my $regex = /(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]/;
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
POSIX
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
Python
regex = re.compile(r'(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]')
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
JavaScript
const regex = /(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]/;
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
PHP
$regex = '/(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]/';
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
Golang
regex, _ := regexp.Compile("(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]")
(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]
Rust
let re = Regex::new(r"(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]").unwrap();
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
C#
var re = new Regex(@"(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]");
(?:0[1-9]|[12]\d|3[01])(?:0[1-9]|1[012])\d{2}[a+-]?\d{3}[\da-z]
Scala
val re = "(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]".r
(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]
Kotlin
val regex = Regex("(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]")
(?:0[1-9]|[12]\\d|3[01])(?:0[1-9]|1[012])\\d{2}[a+-]?\\d{3}[\\da-z]